### PR TITLE
Support arrays at source root by emitting multiple events. Fixes #12

### LINKED ISSUE #12: JSON with root arrays should be handled as multiple events
If JSON with an array as the root is passed in, it should be decoded as separate events.

Example JSON: http://svc.metrotransit.org/NexTrip/VehicleLocations/5?format=json

### PR DESCRIPTION
**Pull request by andrewvc**
Fixes #12

### CODE REVIEW

File: lib/logstash/codecs/json.rb [hunk 1]
```diff
@@ -37,7 +37,15 @@ def register
   def decode(data)
     data = @converter.convert(data)
     begin
-      yield LogStash::Event.new(LogStash::Json.load(data))
+      decoded = LogStash::Json.load(data)
+      if decoded.is_a?(Array)
+        decoded.each {|item|
```
Comment (jordansissel) [inline lib/logstash/codecs/json.rb]: one line with { ... } or use `do ... end` for multiple lines

Comment (andrewvc) [inline lib/logstash/codecs/json.rb]: Ah damn, my lisp is showing.

**Comment by suyograo** [comment]
@andrewvc can we document that this will happen (multiple events if array is supplied) at the top of the file? https://github.com/logstash-plugins/logstash-codec-json/blob/master/lib/logstash/codecs/json.rb#L14

File: lib/logstash/codecs/json.rb (hunk 1 above)
Comment (jordansissel) [inline lib/logstash/codecs/json.rb]: haha, no worries. You should see my ruby after I write C or Javascript. SEMICOLONS EVERYWHERE.

**Comment by jordansissel** [comment]
Code LGTM. There are edge cases (array of something-not-objects), but I think this is a good step forward. Let's add documentation to describe the `[ {...}, {...}, ... ]` support and behavior.

Test code looks good. I haven't run them yet.

**Comment by jordansissel** [comment]
Tests pass for me. Woo! :green_apple:

**Comment by andrewvc** [comment]
@jordansissel cool, will release the new gem then! :rocket:
